test/integration/resource-controller.v2.test.js

```diff
@@ -212,22 +214,34 @@ describe('ResourceControllerV2_integration', () => {
 
     const params = {
       headers: customHeader,
+      limit: resultsPerPage,
+      start: null,
     };
 
     let response;
-    try {
-      response = await service.listResourceInstances(params);
-    } catch (err) {
-      done(err);
-    }
-
-    expect(response).toBeDefined();
-    expect(response.status).toEqual(200);
-    expect(response.result).toBeDefined();
 
-    const result = response.result;
-    expect(result.rows_count).toBeGreaterThanOrEqual(1);
-    expect(result.resources.length).toBeGreaterThanOrEqual(1);
+    do {
+      try {
+        response = await service.listResourceInstances(params);
+      } catch (err) {
+        done(err);
+      }
+      expect(response).toBeDefined();
+      expect(response.status).toEqual(200);
+      expect(response.result).toBeDefined();
+
+      const result = response.result;
+      expect(result.rows_count).toBeGreaterThanOrEqual(1);
+      expect(result.rows_count).toBeLessThanOrEqual(resultsPerPage);
+      expect(result.resources.length).toBeGreaterThanOrEqual(1);
+      expect(result.resources.length).toBeLessThanOrEqual(resultsPerPage);
+
+      if (result.next_url) {
+        params.start = getQueryParam(result.next_url, 'start');
+      } else {
+        params.start = null;
+      }
```

Comment:
```suggestion
      params.start = getQueryParam(result.next_url, 'start');
```
What you have should work fine, but the logic could be simplified a bit.   This same change could be made to each place where you use getQueryParam.